app/src/main/java/com/nervousfish/nervousfish/activities/BluetoothConnectionActivity.java

```diff
@@ -119,15 +123,25 @@ protected void onCreate(final Bundle savedInstanceState) {
      * {@inheritDoc}
      */
     @Override
+    protected void onDestroy() {
+        super.onDestroy();
+        this.serviceLocator.unregisterFromEventBus(this);
+        unregisterReceiver(this.broadcastReceiver);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
     public void onStart() {
         super.onStart();
-
-        this.serviceLocator.registerToEventBus(this);
+        // Get the Paired Devices list
```

Comment:
Unnecessary (the name of the method is `queryPairedDevices`) and possible confusing (`discoveredDevices` doesn't get paired devices`) comment